### DISCUSSION TITLE
Can I update token's POS tag?

### DISCUSSION THREAD

**Discussion by lingvisa** [explosion/spaCy#5743] [Help: Other Questions] [answered]
For some critical POS errors, I want to be able to correct by myself through a dictionary or rules. For dictionary, it could be as simple as:

`dog NN`

Then I can do:

`token.pos_ = 'NN'`

How hard to support such a feature? For serious uses of POS info, this would be very useful!

**Comment by adrianeboyd** [comment, marked as answer]
`token.pos_` is restricted to UPOS tags, so you could do `NOUN` but not `NN`. If you want to use any arbitrary tag like `NN`, use the attribute `token.tag_` instead.

**Comment by lingvisa** [comment]
That's good enough. I vaguely remember spacy token attributes are immutable. Not really, which is good.

**Comment by adrianeboyd** [comment]
Not immutable, just restricted to the UPOS tag set (plus a few custom whitespace-related tags). See the internal list here: https://github.com/explosion/spaCy/blob/27a1cd3c630055802513a20c1e75d0b37943cc39/spacy/parts_of_speech.pxd
